### ISSUE TITLE
Building segmentation ensemble models is slow because of full image inference in the child runs

### ISSUE THREAD

**Issue by ant0nsc** [microsoft/InnerEye-DeepLearning#328]
When building ensemble models, we are usually just interested in the metrics of the final ensemble. But we still do full image inference on validation and test set, before actually starting the ensemble building. We could speed that up by disabling full image inference in the child runs, and only run it with the ensemble model.

[AB#3942](https://innereye.visualstudio.com/60ce1777-00d6-4015-82bc-488a0c00202f/_workitems/edit/3942)

**Comment by ant0nsc** [comment]
- Ensure that we trigger full image inference on training, validation and test set separately. (current: Training set on/off, Val+Test set on/off)
- Trigger full image inference for ensemble model separate from child runs (single checkpoint)
- one solution could be to have a total of 6 flag, we probably want that internally in any case. Externally facing flags could be "smart"

**Comment by JonathanTripp** [comment]
The flags currently are:
-    perform_training_set_inference: bool = \
        param.Boolean(False,
                      doc="If True, run full image inference on the training set at the end of training. If False and "
                          "perform_validation_and_test_set_inference is True (default), only run inference on "
                          "validation and test set. If both flags are False do not run inference.")
 -   perform_validation_and_test_set_inference: bool = \
        param.Boolean(True,
                      doc="If True (default), run full image inference on validation and test set after training.")

In class WorkflowParams
